swap_meet/vendor.py

```diff
@@ -1,2 +1,207 @@
+            False - if old_item is not found inventory will not change
+        '''
+
+        if old_item in self.inventory:
+
+            self.inventory.remove(old_item)
+            return old_item
+
+        else:
+            return False
+        
+    
+    def get_by_id(self, items_id):
+        '''
+        Finds the item in self.inventory with the id = items_id
+        '''
+
+        for item in self.inventory:
+            if items_id == item.id: 
+                return item
+    
+
+    
+    def swap_items(self, other_vendor, my_item, others_item):
+        '''
+    input: 
+        self -
+        other_vendor - object of the same class vendor that self wants to trade with, 
+        my_item - Item self wants to swap, 
+        others_item- item other_vendor wants to swap
+    output: 
+        True - if self and othr_vendor have items that the other wants to swap
+            Side effect- the inventory parameter for self and other_vendor is modified by having the new item and
+                removing the old item
+        False - if self or other_vendor does not have its corrsponding item in their inventory parameter
+        '''
+
+        #Check each of the vendors has the correct item to swap 
+
+        if my_item in self.inventory and others_item in other_vendor.inventory:
+            
+            self.inventory.remove(my_item)
+            other_vendor.inventory.remove(others_item)
+            
+            self.inventory.append(others_item)
+            other_vendor.inventory.append(my_item)
+            
+            return True 
+        
+        # Return False if one of the vendors does not have the item
+        return False
+
+
+    
+    def swap_first_item(self, other_vendor):
+        '''
+        input:
+            self
+            other_vendor - another object of the same class (vendor)
+        output: 
+            False - If either of the vendors have empty inventories 
+            True - If both of the vendors have at least one item in their inventories 
+                and a swap is done 
+        '''
+
+        if len(self.inventory) and len(other_vendor.inventory):
+
+            my_item = self.inventory[0]
+            their_item = other_vendor.inventory[0]
+            self.swap_items(other_vendor,my_item,their_item)
+
+            return True 
+
+        return False 
+    
+    
+
+    def get_by_category(self, category):
+        '''
+        input:
+            self -
+            category - str specifiying which type of item the inventory of self is going to be filtered by 
+        output: 
+            items_in_category - list of items that self has in its inventory under the category
+                specified as an argument
+        '''
+
+        items_in_category = []
+
+        for item in self.inventory:
+
+            if item.get_category() == category:
+                items_in_category.append(item)
+
+        return items_in_category
```

Comment:
Notice this pattern of: 
```python
result_list = []
for element in source_list:
    if some_condition(element):
        result_list.append(element)
```

can be rewritten using a list comprehension as:
```python
result_list = [element for element in source_list if some_condition(element)]
```

Which here would look like:
```python
items_in_category = [item for item in self.inventory if item.get_category() == category]
return items_in_category
```

At first, this may seem more difficult to read, but comprehensions are a very common python style, so I encourage y’all to try working with them!